native/cocos/engine/Engine.cpp

```diff
@@ -311,7 +308,6 @@ int32_t Engine::restartVM() {
     cc::EventDispatcher::init();
     CC_CURRENT_APPLICATION()->init();
 
-    cc::gfx::DeviceManager::addSurfaceEventListener();
```

Comment:
This line seems necessary since new EventDispatcher init above.

Reply:
Yep, you are right.